### PR TITLE
Region planet seed

### PR DESCRIPTION
**Pull request by pdeguing**

### CODE REVIEW

File: templates/leftbar.html [hunk 1]
```diff
@@ -1,55 +1,19 @@
...
+	{{ range .Data.Planets}}
+
+	<a class="panel-block is-active" href="/planet/{{ .ID }}/">
+			<figure class="image is-48x48 Planet-menu__icon">
+				<img src="/static/images/planet-icons/{{ .PlanetSkin }}.png" class="is-rounded" alt="planet image" width="48px"
+					height="48px">
+			</figure>
+			<div>
+				<div>{{ .Name }}</div>
+				<div>[0:30:3:0]</div>
```
Comment (MarkKremer) [inline templates/leftbar.html]: This can be dynamic.

File: templates/planet.overview.html [hunk 2]
```diff
@@ -17,7 +17,7 @@ <h1 class="title dashboard-title">Avalon</h1>
         </div>
         <div class="columns is-inline-flex is-gapless dashboard-item">
           <span class="column is-half is-growing dashboard-item-name">Planet type</span>
-          <span class="column is-narrow dashboard-item-value">habitable</span>
+		  <span class="column is-narrow dashboard-item-value">{{ .Data.Planet.PlanetType }}</span>
```
Comment (MarkKremer) [inline templates/planet.overview.html]: Gas giant is displayed as `gas_giant`.

File: planet_helpers.go [hunk 3]
```diff
@@ -25,8 +25,12 @@ func userPlanet(r *http.Request, tx *ent.Tx) (*ent.Planet, error) {
...
 	p, err := tx.Planet.
 		Query().
-		Where(planet.HasOwnerWith(user.IDEQ(u.ID))).
-		Offset(n - 1).
+		Where(
+			planet.And(
+				planet.HasOwnerWith(user.IDEQ(u.ID)),
+				planet.ID(n),
+			),
+		).
```
Comment (MarkKremer) [inline planet_helpers.go]: Figured out the formatting. 👍

File: data/seed.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+		SetOrbitCode(orbit).
+		SetSuborbitCode(suborbit).
+		SetPositionCode(positionCode).
+		SetPlanetSkin(planetSkin).
+		SaveX(context.Background())
+
+}
+
+func generatePlanet(r *rand.Rand, planetSkins []os.FileInfo, region int, system int, orbit int, suborbit int) bool {
+	n := r.Intn(9)
```
[the remark this answers is not in the document]
Comment (pdeguing) [reply data/seed.go]: Ahah, it's just to set the chance of generation to 4/10.

File: planet_helpers.go [hunk 5]
```diff
@@ -45,13 +49,59 @@ func userPlanet(r *http.Request, tx *ent.Tx) (*ent.Planet, error) {
...
+}
+
+// regionPlanets retrieves all planets
+func regionPlanets(w http.ResponseWriter, r *http.Request) ([]*ent.Planet, error) {
+	p, err := data.Client.Planet.Query().
+		Order(ent.Asc(planet.FieldPositionCode)).
+		All(r.Context())
+
+	if _, ok := err.(*ent.ErrNotFound); ok {
+		return nil, newNotFoundError(err)
```
[the remark this answers is not in the document]
Comment (pdeguing) [reply planet_helpers.go]: Changed, let me know if I did it the properway

[the remark this answers is not in the document]
Comment (MarkKremer) [reply planet_helpers.go]: I think you didn't push it yet.

File: route_dashboard.go [hunk 6]
```diff
@@ -2,18 +2,45 @@ package main
...
 
 // GET /dashboard/cartography
 // Show map page
 func serveCartography(w http.ResponseWriter, r *http.Request) {
-	generateHTML(w, r, "cartography", nil, "layout", "private.navbar", "dashboard", "leftbar", "cartography")
+	p, err := regionPlanets(w, r)
+	if err != nil {
+		serveError(w, r, err)
+	}
+	generateHTML(w, r, "cartography", p, "layout", "private.navbar", "dashboard", "leftbar", "cartography")
```
[the remark this answers is not in the document]
Comment (pdeguing) [reply route_dashboard.go]: I struggle finding a proper way to manage this. Also, ultimately if we have a notifications bar on the right, we would need timer data on most views. So we could reuse planetViewData in most cases.

File: planet_helpers.go [hunk 7]
```diff
@@ -72,7 +72,8 @@ func regionPlanets(w http.ResponseWriter, r *http.Request) ([]*ent.Planet, error
 		All(r.Context())
 
 	if _, ok := err.(*ent.ErrNotFound); ok {
-		return nil, newNotFoundError(err)
+		p := make([]*ent.Planet, 0)
+		return p, newNotFoundError(err)
```
Comment (MarkKremer) [inline planet_helpers.go]: Rereading this, I don't think `All(r.Context)` actually returns a not found error. So this whole if block can be removed.

But if it did, what I meant was:
```golang
return nil, nil
```
So you're returning a nil-slice (which Go handles as if you passed an empty slice) and no error.

File: data/seed.go [hunk 8]
```diff
@@ -27,14 +26,13 @@ func randomPlanetType(r *rand.Rand) planet.PlanetType {
...
+func getPositionCode(region, system, orbit, suborbit int) int {
+	return suborbit + orbit << 4 + system << 8 + region << 12
 
 }
 
-func randomPlanetName() string {
-	seed := time.Now().UTC().UnixNano()
-	nameGenerator := namegenerator.NewNameGenerator(seed)
+func randomPlanetName(r *rand.Rand) string {
+	nameGenerator := namegenerator.NewNameGenerator(int64(r.Intn(9)))
```
[the remark this answers is not in the document]
Comment (pdeguing) [reply data/seed.go]: I don't think so, the argument in the seed. Anyway this is temporary until I build our name generator.

File: planet_helpers.go (hunk 7 above)
Comment (pdeguing) [reply planet_helpers.go]: Ok

**Pull request merged**